http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -329,13 +340,20 @@ protected AuthChallenge checkStateCookie() {
         // reset the cookie
         log.debug("** reseting application state cookie");
         facade.getResponse().resetCookie(deployment.getStateCookieName(), stateCookie.getPath());
-        String stateCookieValue = getCookieValue(deployment.getStateCookieName());
+        String cookiesStr = getCookieValue(deployment.getStateCookieName());
 
         String state = getQueryParamValue(facade, STATE);
         if (state == null) {
             log.warn("state parameter was null");
             return challenge(HttpStatus.SC_BAD_REQUEST, AuthenticationError.Reason.INVALID_STATE_COOKIE, null);
         }
+
+        String stateCookieValue = cookiesStr;
+        if (cookiesStr != null) {
+            String[] cookieArr = cookiesStr.split(";");
+            stateCookieValue = cookieArr[0];
+        }
```

Comment:
This shouldn't be needed.

Reply:
Changes to AbstractBaseHttpTest resolved this